google/cloud/documentai_toolbox/wrappers/document.py

```diff
@@ -38,6 +39,37 @@
 from google.cloud.documentai_toolbox.wrappers.page import FormField, Page
 
 
+def _chunks_from_shards(
+    shards: List[documentai.Document],
+) -> List[documentai.Document.ChunkedDocument.Chunk]:
+    return [chunk for shard in shards for chunk in shard.chunked_document.chunks]
+
+
+def _document_layout_blocks_from_shards(
+    shards: List[documentai.Document],
+) -> List[documentai.Document.DocumentLayout.DocumentLayoutBlock]:
+    def extract_blocks(
+        blocks: List[documentai.Document.DocumentLayout.DocumentLayoutBlock],
+    ) -> List[documentai.Document.DocumentLayout.DocumentLayoutBlock]:
+        unnested_blocks = []
+        queue = collections.deque(blocks)
+
+        while queue:
+            block = queue.popleft()
+            unnested_blocks.append(block)
+            # Add the nested blocks to the stack in the correct order
+            if block.text_block and block.text_block.blocks:
+                queue.extendleft(reversed(block.text_block.blocks))
+
+        return unnested_blocks
+
+    return [
+        block
+        for shard in shards
+        for block in extract_blocks(shard.document_layout.blocks)
+    ]
```

Comment:
Instead of a list, perhaps an iterator that traverses the graph when the user accesses it?  This way in the querying use case the user can stop iterating as soon as the query has been matched, without having to pay the cost of having the whole graph of blocks flattened.